src/routes/docs/tutorials/apple/step-5/+page.markdoc

```diff
@@ -0,0 +1,402 @@
+---
+layout: tutorial
+title: Manage navigation 
+description: Add navigation to your SwiftUI app with Appwrite authentication.
+step: 5
+---
+
+# Router {% #router %}
+
+Add the following code to Router.swift File in the root folder
+
+```swift
+import SwiftUI
+
+
+enum Tabs {
+    case home, progress, setting
+}
+
+enum Route {
+    case appwrite
+    case home
+    case auth
+    case add_habit
+}
+
+extension Route: View {
+    var body: some View {
+        switch self {
+        case .appwrite:
+             ContentView()
+        case .home:
+            HomeView()
+        case .auth:
+            AuthScreen()
+        case .add_habit:
+            AddHabitScreen()
+        }
+        
+    }
+}
+
+
+extension Route: Hashable {
+    static func == (lhs: Route, rhs: Route) -> Bool {
+        return lhs.compareString == rhs.compareString
+    }
+    
+    var compareString: String {
+        switch self {
+        case .appwrite:
+            return "appwrite"
+        case .home:
+            return "home"
+        case .auth:
+            return "auth"
+        case .add_habit:
+            return "addHabit"
+        }
+    }
+}
+
+
+final class Router: ObservableObject {
+    @Published var routes = [Route]()
+    @Published var selectedTab: Tabs = .home
+    
+    func push(_ screen: Route) {
+        routes.append(screen)
+    }
+    
+    func pushReplacement(_ screen: Route) {
+        if routes.isEmpty {
+            routes.append(screen)
+        } else {
+            routes[routes.count - 1] = screen
+        }
+    }
+    
+    func pop() {
+        routes.removeLast()
+    }
+    
+    func popUntil(predicate: (Route) -> Bool) {
+        if let last = routes.popLast() {
+            guard predicate(last) else {
+                popUntil(predicate: predicate)
+                return
+            }
+        }
+    }
+    
+    func reset() {
+        routes = []
+    }
+}
+```
+
+# ContentView {% #contentview %}
+
+Update the ContentView.swift to reflect some changes related to our navigation
+
+```swift
+import SwiftUI
+
+struct ContentView: View {
+    
+    @EnvironmentObject private var router: Router
```

Comment:
I'm getting a /Users/wen/Documents/GitHub/tracker/tracker/ContentView.swift:7:6 Generic struct 'EnvironmentObject' requires that 'SnackBarService' conform to 'ObservableObject'

here